python/paddle/distributed/auto_parallel/context.py

```diff
@@ -0,0 +1,376 @@
+            tensor_node_id] = tensor_dist_attr
+
+    def get_op_distributed_attr_for_graph(self, op_node):
+        op_node_id = op_node.id()
+        op_dist_attr = self._op_distributed_attr_map_for_graph.get(op_node_id,
+                                                                   None)
+        return op_dist_attr
+
+    def set_op_distributed_attr_for_graph(self, op_node, op_dist_attr):
+        op_node_id = op_node.id()
+        self._op_distributed_attr_map_for_graph[op_node_id] = op_dist_attr
+
+    def initialize_distributed_attr_for_program(self, program):
+        if self._is_initialized_for_program:
+            return
+        for block in program.blocks:
+            for tensor in block.vars.values():
+                # Since only tensors have distributed attributes, it's better to make sure var is a tensor
+                tensor_dist_attr = self.get_tensor_distributed_attr_for_program(
+                    tensor)
+                if tensor_dist_attr is None:
+                    tensor_dist_attr = TensorDistributedAttribute(tensor, self)
+                    self._copy_distributed_attr_from_tensor_desc(
+                        tensor.desc, tensor_dist_attr)
+                    self.set_tensor_distributed_attr_for_program(
+                        tensor, tensor_dist_attr)
+                tensor_dist_attr.set_shape(tensor.desc.shape())
+                if tensor_dist_attr.get_process_mesh() is not None:
+                    tensor_dist_attr.mark_as_annotated("process_mesh")
+                if tensor_dist_attr.get_dims_mapping() is None:
+                    tensor_dims_mapping = [
+                        -1 for _ in range(len(tensor.desc.shape()))
+                    ]
+                    tensor_dist_attr.set_dims_mapping(tensor_dims_mapping)
+                else:
+                    tensor_dist_attr.mark_as_annotated("dims_mapping")
+                if isinstance(tensor, framework.Parameter):
+                    tensor_dist_attr.mark_as_parameter()
+            for op in block.ops:
+                op_dist_attr = self.get_op_distributed_attr_for_program(op)
+                if op_dist_attr is None:
+                    op_dist_attr = OperatorDistributedAttribute(op, self)
+                    self._copy_distributed_attr_from_op_desc(op.desc,
+                                                             op_dist_attr)
+                    self.set_op_distributed_attr_for_program(op, op_dist_attr)
+                # Default distributed implementation for all operators
+                # This will be updated during the completion prcess
+                op_dist_attr.set_impl_idx(-2)
+                if op_dist_attr.get_process_mesh() is not None:
+                    op_dist_attr.mark_as_annotated("process_mesh")
+                for tensor_name in op.input_arg_names:
+                    # There may be a better way to find the tensor by name
+                    tensor = op.block._var_recursive(tensor_name)
```

Comment:
try op.block.vars[tensor_name]?

Reply:
Try to make sure to find the tensor from current block level up to the top level (unnecessarily now).